### PR TITLE
Add support for whitelisting ecosystems.

### PR DESCRIPTION
**Pull request by deadly-panda**

### CODE REVIEW

File: agent/osv_agent.py [hunk 1]
```diff
@@ -171,17 +173,17 @@ def _process_fingerprint_file(self, message: m.Message) -> None:
         path = message.data.get("path")
 
         if package_version is None:
-            logger.error("Error: Version must not be None.")
```
Comment (elyousfi5) [inline agent/osv_agent.py:174]: shouldnt this be just warning or debug ?

Comment (deadly-panda) [reply agent/osv_agent.py:174]: because it's none of those, it's not really a problem.

File: agent/osv_output_handler.py [hunk 2]
```diff
@@ -32,6 +32,12 @@
     "POTENTIALLY": 5,
 }
 
+OSV_WHITELISTED_ECOSYSTEM = {
+    "ELF_LIBRARY": ["OSS-Fuzz", "Alpine", "Debian", "Linux"],
+    "MACHO_LIBRARY": ["OSS-Fuzz", "Alpine", "Debian", "Linux", "SwiftURL"],
```
Comment (3asm) [inline agent/osv_output_handler.py:37]: Should be removed.

File: agent/osv_agent.py [hunk 3]
```diff
@@ -171,22 +175,30 @@ def _process_fingerprint_file(self, message: m.Message) -> None:
...
-        api_result = osv_service_api.query_osv_api(
-            package_name=package_name,
-            version=package_version,
-            ecosystem=OSV_ECOSYSTEM_MAPPING.get(str(package_type)),
-        )
-        if api_result is None or api_result == {}:
-            return None
+        all_vulnerabilities = []
+        ecosystems = OSV_ECOSYSTEM_MAPPING.get(str(package_type))
+        if ecosystems is not None and len(ecosystems) > 0:
```
Comment (3asm) [inline agent/osv_agent.py:185]: How about if there only one, where is the logic for that?

Comment (deadly-panda) [reply agent/osv_agent.py:185]: I made all the single values into lists of one element.

File: agent/osv_output_handler.py [hunk 4]
```diff
@@ -216,7 +228,7 @@ def parse_vulnerabilities_osv_api(
         if old_risk is not None and new_risk is not None and old_risk > new_risk:
             highest_risk_vuln_info["risk"] = risk
             highest_risk_vuln_info["cvss_v3_vector"] = _get_cvss_v3_vector(
-                vulnerability.get("severity")
+                vulnerability.get("severity", [])
```
Comment (oussamaessaji) [inline agent/osv_output_handler.py:231]: There is a check for `severity_data is not None` in `_get_cvss_v3_vector` can you remove it

File: agent/osv_agent.py [hunk 5]
```diff
@@ -171,17 +175,25 @@ def _process_fingerprint_file(self, message: m.Message) -> None:
...
-            logger.error("Error: Package name must not be None.")
+            logger.warning("Error: Package name must not be None.")
             return None
 
+        ecosystems = OSV_ECOSYSTEM_MAPPING.get(str(package_type))
+        whitelisted_ecosystems = None
+        ecosystem = None
+        if isinstance(ecosystems, str):
+            ecosystem = ecosystems
+        elif isinstance(ecosystems, list):
```
Comment (3asm) [inline agent/osv_agent.py:188]: Man, just keep it as a list with a single element and check the count!

File: agent/osv_output_handler.py [hunk 6]
```diff
@@ -195,16 +199,24 @@ def parse_vulnerabilities_osv_api(
     highest_risk_vuln_info: dict[str, str] = {}
     if len(vulnerabilities) == 0:
         return []
-    for vulnerability in vulnerabilities:
+
+    whitlisted_vulnerabilities = _whitelist_vulnz_from_ecosystems(
```
Comment (3asm) [inline agent/osv_output_handler.py:203]: If there is one ecosystem, pass in the query!

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**